### PR TITLE
adding environment variable option when treating flags

### PR DESCRIPTION
**Pull request by jimenez**
Signed-off-by: Isabel Jimenez <contact.isabeljimenez@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @aluzzardi @abronan this is needed from mesos.

**Comment by abronan** [comment]
LGTM

File: cluster/options.go [hunk 1]
```diff
@@ -9,19 +11,22 @@ import (
...
 // String returns a string from the driver options
-func (do DriverOpts) String(key string) (string, bool) {
+func (do DriverOpts) String(key, env string) (string, bool) {
 	for _, opt := range do {
 		kv := strings.SplitN(opt, "=", 2)
 		if kv[0] == key {
 			return kv[1], true
 		}
 	}
+	if env := os.Getenv(env); env != "" {
```
Comment (aluzzardi) [inline cluster/options.go]: Why not `os.Getenv("SWARM_CLUSTER_OPT_" + strings.ToUpper(key))` so you don't need to pass an extra parameter?

Comment (jimenez) [inline cluster/options.go]: Do we want all the flags to be accessible via the environment? For example in this PR swarm.overcommit could only be set via flag.

Comment (aluzzardi) [inline cluster/options.go]: Uhm, you are right. We could replace `.` with `_` so it'd be `SWARM_CLUSTER_OPT_SWARM_OVERCOMMIT` but maybe that's overkill?

**Comment by aluzzardi** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**